### PR TITLE
Add kibit.driver/exec to support call via `clojure -X:`

### PR DESCRIPTION
**Pull request by carrete**
A follow-up to https://github.com/clj-commons/kibit/issues/221. I hope the documentation included in this pull-request is sufficient enough to explain this pull-request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,10 +140,16 @@ instead of:
 Add the following to your aliases
 
 ```clojure
-:kibit {:extra-deps {jonase/kibit {:mvn/version "0.1.8"}}
-         :main-opts ["-e" "(require,'[kibit.driver,:as,k])(k/external-run,[\"src\"],nil)"]}
+:kibit {:extra-deps {jonase/kibit {:mvn/version "0.2.0"}}
```
Comment (carrete) [inline README.md:143]: I made the assumption that the next release will be `0.2.0`. Please let me know if this needs to be changed

File: src/kibit/driver.clj [hunk 2]
```diff
@@ -84,3 +84,62 @@
...
     (System/exit 0)
     (System/exit 1)))
+
+(defn exec
+  "Given a [Clojure CLI-style map][Execute a function] `options`, turn this map
+  into an equivalent set of options as expected by run and external-run above.
+
+  Please note that rules is not supported. nil is passed to external-run below
+  which enables all rules (see kibit.rules/all-rules). I don't know what rules
+  should be. A list of rule names as strings doesn't seem to be it.
```
Comment (carrete) [inline src/kibit/driver.clj:94]: I'd be happy to support rules in this pull-request. Could someone tell me what this should look like on the command-line?

File: README.md (hunk 1 above)
Comment (NoahTheDuke) [reply README.md:143]: I don't think this is necessary. We can update all of the version references as one when it comes time to cut a new release.

File: src/kibit/driver.clj (hunk 2 above)
Comment (NoahTheDuke) [reply src/kibit/driver.clj:94]: Kibit currently doesn't support rule names.

File: README.md (hunk 1 above)
Comment (carrete) [reply README.md]: Cool. I added placeholders instead to be updated at the time of the next release.

**Comment by carrete** [comment]
@NoahTheDuke Are there any more changes you'd like me to make? Thanks

**Comment by NoahTheDuke** [comment]
Nope, thanks for the reminder.

**Pull request merged**